jrugged-core/src/main/java/org/fishwife/jrugged/SkepticBreaker.java

```diff
@@ -0,0 +1,471 @@
+     */
+    public void invoke(Runnable r) throws Exception {
+        if (!byPass) {
+            if (!allowRequest()) {
+            	if(!isHardTrip){
+            		try{
+            			Runnable mockRunnable = createMock(Runnable.class);
+            		    mockRunnable.run(); 
+            		}
+            		catch (Throwable cause){
+            			handleFailure(cause); 
+            		}
+            	}
+                throw mapException(new BreakerException());
+            }
+
+            try {
+                r.run();
+                close();
+                return;
+            } catch (Throwable cause) {
+                handleFailure(cause);
+            }
+            throw new IllegalStateException("not possible");
+        }
+        else {
+            r.run();
+        }
+    }
+
+    /** Wrap the given service call with the {@link SkepticBreaker}
+     *  protection logic.
+     *  @param r the {@link Runnable} to attempt
+     *  @param result what to return after <code>r</code> succeeds
+     *  @return result
+     *  @throws BreakerException if the
+     *    breaker was OPEN and this attempt wasn't the
+     *    reset attempt
+     *  @throws Exception if <code>c</code> throws one during
+     *    execution
+     */
+    public <V> V invoke(Runnable r, V result) throws Exception {
+        if (!byPass) {
+            if (!allowRequest()) {
+            	if(!isHardTrip){
+            		try{
+            			Runnable mockRunnable = createMock(Runnable.class);
+            		    mockRunnable.run(); 
+            		}
+            		catch (Throwable cause){
+            			handleFailure(cause); 
+            		}
+            	}
+                throw mapException(new BreakerException());
+            }
+
+            try {
+                r.run();
+                close();
+                return result;
+            } catch (Throwable cause) {
+                handleFailure(cause);
+            }
+            throw new IllegalStateException("not possible");
+        }
+        else {
+            r.run();
+            return result;
+        }
+    }
+
+    /**
+     * Causes the {@link SkepticBreaker} to trip and OPEN; no new
+     *  requests will be allowed until the <code>SkepticBreaker</code>
+     *  resets. Skeptic level increases and timers are updated accordingly.
+     */
+    public void trip() {
+    	if (state != BreakerState.OPEN) {
+            openCount.getAndIncrement();
+        }
+    	if (skepticLevel.get() < maxLevel.get()) {
+    		increaseSkepticLevel();
+    	}
+        state = BreakerState.OPEN;
+        lastFailure.set(System.currentTimeMillis());
+
+        notifyBreakerStateChange(getStatus());
+    }
+    
+    /**
+     * Increments the skeptic level and updates timers; occurs when the 
+     * {@link SkepticBreaker} trips and moves from CLOSED to OPEN
+     */
+    public void increaseSkepticLevel() {
+    	skepticLevel.set(skepticLevel.incrementAndGet());
+    	updateTimers();
```

Comment:
Why wouldn't I have this call immediately follow the increaseSkepticLevel() call in the method above inside the if block? It seems like an unexpected result to have a call to something called increaseSkepticLevel be also updating timers.